File: YardimUygulamaJava/app/src/main/java/com/example/yardimuygulamajava/repo/PatientRepo.java
package com.example.yardimuygulamajava.repo;

import com.example.yardimuygulamajava.model.HelpCancelBody;
import com.example.yardimuygulamajava.model.HelpConfirmBody;
import com.example.yardimuygulamajava.model.HelpCreateBody;
import com.example.yardimuygulamajava.model.HelpRequestActive;
import com.example.yardimuygulamajava.service.ApiClient;
import com.example.yardimuygulamajava.service.ApiOk;
import com.example.yardimuygulamajava.service.ApiService;

import retrofit2.Call;

public class PatientRepo {
    private final ApiService api = ApiClient.api();

    public Call<ApiOk<Object>> createHelp(long patientId, String servis, String oda, double lat, double lng) {
        return api.createHelp(new HelpCreateBody(patientId, servis, oda, lat, lng));
    }

    public Call<ApiOk<HelpRequestActive>> myActive(long patientId) {
        return api.myActive(patientId);
    }

    public Call<ApiOk<Object>> confirm(long requestId, long patientId) {
        return api.confirm(new HelpConfirmBody(requestId, patientId));
    }

    public Call<ApiOk<Object>> cancel(long requestId, long patientId) {
        return api.cancel(new HelpCancelBody(requestId, patientId));
    }
}

File: YardimUygulamaKotlin/app/src/main/java/com/example/yardimuygulamakotlin/repo/PatientRepo.kt
package com.example.yardimuygulamakotlin.repo

import com.example.yardimuygulamakotlin.model.ApiOk
import com.example.yardimuygulamakotlin.model.HelpCancelBody
import com.example.yardimuygulamakotlin.model.HelpConfirmBody
import com.example.yardimuygulamakotlin.model.HelpCreateBody
import com.example.yardimuygulamakotlin.model.HelpRequestActive
import com.example.yardimuygulamakotlin.service.ApiClient
import com.example.yardimuygulamakotlin.service.ApiService

// PatientRepo.kt
class PatientRepo(private val api: ApiService = ApiClient.api) {

    suspend fun createHelp(
        patientId: Long,
        servis: String,
        oda: String,
        lat: Double,
        lng: Double,
        hastane: String? = null
    ): ApiOk<Any>? {
        val body = HelpCreateBody(
            patient_id = patientId,
            hastane_adi = hastane,
            servis_adi = servis,
            oda_no = oda,
            lat = lat,
            lng = lng
        )
        val r = api.helpCreate(body)
        return if (r.isSuccessful) r.body() else null
    }
    suspend fun cancel(requestId: Long, patientId: Long): ApiOk<Any>? {
        val r = api.cancel(HelpCancelBody(requestId, patientId))
        return if (r.isSuccessful) r.body() else null
    }
    suspend fun myActive(patientId: Long): ApiOk<HelpRequestActive>? {
        val r = api.myActive(patientId)
        return if (r.isSuccessful) r.body() else null
    }

    suspend fun confirm(requestId: Long, patientId: Long): ApiOk<Any>? {
        val r = api.confirm(HelpConfirmBody(requestId, patientId))
        return if (r.isSuccessful) r.body() else null
    }
}
